### PR TITLE
[TASK] Adjust TCA for type=password examples

### LINKED ISSUE #309: Adjust TCA to use type=password
See https://forge.typo3.org/issues/97104 and https://review.typo3.org/c/Packages/TYPO3.CMS/+/73832 for reference.

### PR DESCRIPTION
**Pull request by o-ba**
Closes #309

### CODE REVIEW

File: Classes/TcaDataGenerator/FieldGenerator/TypePassword.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
```
Comment (simonschaufi) [inline Classes/TcaDataGenerator/FieldGenerator/TypePassword.php:3]: Empty line after declare

File: Classes/Service/KauderwelschService.php [hunk 2]
```diff
@@ -51,6 +51,16 @@ public function getWord(): string
         return 'lipsum';
     }
 
+    /**
+     * Get a single password
+     *
+     * @return string
```
Comment (simonschaufi) [inline Classes/Service/KauderwelschService.php:57]: can be dropped

File: Classes/TcaDataGenerator/FieldGenerator/TypePassword.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+use TYPO3\CMS\Styleguide\TcaDataGenerator\FieldGeneratorInterface;
+
+/**
+ * Generate data for type=password fields
+ */
+class TypePassword extends AbstractFieldGenerator implements FieldGeneratorInterface
+{
+    /**
+     * @var array General match if type=password
```
Comment (simonschaufi) [inline Classes/TcaDataGenerator/FieldGenerator/TypePassword.php:27]: Comment must be above `@var` declaration

File: Classes/TcaDataGenerator/FieldGenerator/TypePassword.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            'config' => [
+                'type' => 'password',
+            ],
+        ],
+    ];
+
+    /**
+     * Returns the generated value to be inserted into DB for this field
+     *
+     * @param array $data
```
Comment (simonschaufi) [inline Classes/TcaDataGenerator/FieldGenerator/TypePassword.php:40]: can be dropped

**Review by simonschaufi — CHANGES_REQUESTED** [review]
please adjust code style

File: Classes/TcaDataGenerator/FieldGenerator/TypePassword.php (hunk 1 above)
Comment (o-ba) [reply Classes/TcaDataGenerator/FieldGenerator/TypePassword.php:3]: There is no such rule in the coding style rules.

**Review by simonschaufi — APPROVED** [review]

**Review by sbuerk — APPROVED** [review]

File: Classes/TcaDataGenerator/FieldGeneratorResolver.php [hunk 5]
```diff
@@ -64,6 +64,9 @@ class FieldGeneratorResolver
         // General type=link generator
         FieldGenerator\TypeLink::class,
 
+        // General type=link generator
+        FieldGenerator\TypeLink::class,
```
[the remark this answers is not in the document]
Comment (o-ba) [reply Classes/TcaDataGenerator/FieldGeneratorResolver.php]: Done.

File: Classes/TcaDataGenerator/FieldGenerator/TypePassword.php [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+            ],
+        ],
+    ];
+
+    /**
+     * Returns the generated value to be inserted into DB for this field
+     */
+    public function generate(array $data): string
+    {
+        return $this->kauderwelschService->getPassword();
```
Comment (lolli42) [inline Classes/TcaDataGenerator/FieldGenerator/TypePassword.php:44]: hmm. the issue is there that 'hashed' => true is not considered for the default-language record. maybe we need two generators here: one that understands that the value in db has to be hashed
... i'll see what i can do.

**Pull request merged**